stacker/plan.py

```diff
@@ -206,6 +206,8 @@ def build_graph(steps):
         for dep in step.requires:
             graph.connect(step, dep)
 
+    graph.transitive_reduction()
```

Comment:
Curious why this was added to every build_graph call?  Was it meant to provide better performance?

Reply:
Technically, performing the transitive reduction produces an identical graph, but, it's really not necessary to do it here, so I'm probably gonna pull it out of this.

We only really want to do it for the `dot` format, so that going to a png doesn't produce a mess. Maybe it should be a flag, since I can see times where you might want to see all edges, and not just the simplified version.

Reply:
Yeah, I figured it was only necessary for the dot format.  I think just using it there is probably fine, though you're far more knowledgeable about how the graph is used than me.

Reply:
Made this change. The default is to not perform a reduction, and `--reduce` can be passed to generate a "simpler" graph.